packages/rest/src/utils/agent.ts

```diff
@@ -1,79 +1,196 @@
+  extraAnonCredsRegistries?: AnonCredsRegistry[]
+  multiTenant: boolean
+}) => {
+  const legacyIndyCredentialFormatService = new LegacyIndyCredentialFormatService()
+  const legacyIndyProofFormatService = new LegacyIndyProofFormatService()
+
+  const baseModules = {
+    connections: new ConnectionsModule({
+      autoAcceptConnections: options.autoAcceptConnections,
+    }),
+    proofs: new ProofsModule({
+      autoAcceptProofs: options.autoAcceptProofs,
+      proofProtocols: [
+        new V1ProofProtocol({
+          indyProofFormat: legacyIndyProofFormatService,
+        }),
+        new V2ProofProtocol({
+          proofFormats: [legacyIndyProofFormatService, new AnonCredsProofFormatService()],
+        }),
+      ],
+    }),
+    credentials: new CredentialsModule({
+      autoAcceptCredentials: options.autoAcceptCredentials,
+      credentialProtocols: [
+        new V1CredentialProtocol({
+          indyCredentialFormat: legacyIndyCredentialFormatService,
+        }),
+        new V2CredentialProtocol({
+          credentialFormats: [legacyIndyCredentialFormatService, new AnonCredsCredentialFormatService()],
+        }),
+      ],
+    }),
+    anoncreds: new AnonCredsModule({
+      registries: [new IndyVdrAnonCredsRegistry(), ...(options.extraAnonCredsRegistries ?? [])],
+      anoncreds,
+    }),
+    askar: new AskarModule({
+      ariesAskar,
+      multiWalletDatabaseScheme: AskarMultiWalletDatabaseScheme.ProfilePerWallet,
+    }),
+    mediator: new MediatorModule({
+      autoAcceptMediationRequests: options.autoAcceptMediationRequests,
+    }),
+    dids: new DidsModule({
+      registrars: [new KeyDidRegistrar(), new JwkDidRegistrar(), new PeerDidRegistrar(), new IndyVdrIndyDidRegistrar()],
+      resolvers: [
+        new WebDidResolver(),
+        new KeyDidResolver(),
+        new JwkDidResolver(),
+        new PeerDidResolver(),
+        new IndyVdrIndyDidResolver(),
+        new IndyVdrSovDidResolver(),
+      ],
+    }),
+  } as const
+
+  const modules: typeof baseModules & { tenants?: TenantsModule<typeof baseModules>; indyVdr?: IndyVdrModule } =
+    baseModules
+
+  if (options.multiTenant) {
+    modules.tenants = new TenantsModule({
+      sessionLimit: Infinity,
+    })
+  }
+
+  if (options.indyLedgers) {
+    modules.indyVdr = new IndyVdrModule({
+      indyVdr,
+      networks: options.indyLedgers,
+    })
+  }
+
+  return modules
+}
 
 export const setupAgent = async ({
   name,
-  publicDidSeed,
   endpoints,
-  port,
+  extraAnonCredsRegistries,
+  httpInboundTransportPort,
+  multiTenant = false,
 }: {
   name: string
-  publicDidSeed: string
   endpoints: string[]
-  port: number
-}) => {
-  const logger = new TsLogger(LogLevel.debug)
+  extraAnonCredsRegistries?: AnonCredsRegistry[]
+  httpInboundTransportPort?: number
+  multiTenant?: boolean
+}): Promise<RestRootAgent | RestRootAgentWithTenants> => {
+  // FIXME: logger should not be enabled by default
+  const logger = new TsLogger(LogLevel.off)
```

Comment:
Would this be possible to set with the start script? with something like `-vvv` for loglevel=3?

Reply:
Yes we can do it for the CLI part already --log-level. I need to improve the non-cli (import rest server) api a bit. Will look at it in my next PR if okay